### ISSUE TITLE
Unable to create context with no statement limit from an engine which has previously had a statement limit set

### ISSUE THREAD

**Issue by jdimona** [oracle/graal#2974]
**Describe GraalVM and your environment :**
 - GraalVM version or commit id if built from source: 20.2.0-dev
 - CE or EE: CE
 - JDK version: JDK8
 - OS and OS Version: macOS Catalina
 - Architecture: amd64
 - The output of `java -Xinternalversion`:
```
OpenJDK 64-Bit Server VM GraalVM CE 20.2.0-dev (25.252-b09-jvmci-20.2-b02) for bsd-amd64 JRE (1.8.0), built on Jun 29 2020 01:12:29 by "graal" with gcc 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)
```

**Describe the issue**
Once a context has been created from an engine with a statement limit, it is not possible to create a new context from that engine without a statement limit. When a new context with no limit is created an IllegalArgumentException is thrown.

**Code snippet or code repository that reproduces the issue**
```
// Groovy
@Test
void "Cannot unset statement limit once established"() {
    Engine engine = Engine.newBuilder().build()
    ResourceLimits limits = ResourceLimits.newBuilder()
            .statementLimit(1000, { s -> !s.isInternal() })
            .build()
    Context context = Context.newBuilder("js")
            .engine(engine)
            .resourceLimits(limits)
            .build()
    context.close(true)

    Context newContext = Context.newBuilder("js").engine(engine).build()
}
```

**Expected behavior**
I would expect there to be some way to create a new context without a limit since it is possible to go from not having a limit -> any limit -> any other limit.

<details><summary>Details</summary>

```
java.lang.IllegalArgumentException: Using multiple source predicates per engine is not supported. The same statement limit source predicate must be used for all polyglot contexts that are assigned to the same engine. Resolve this by using the same predicate instance when constructing the limits object with ResourceLimits.Builder.statementLimit(long, Predicate).

	at com.oracle.truffle.polyglot.PolyglotEngineException.illegalArgument(PolyglotEngineException.java:128)
	at com.oracle.truffle.polyglot.PolyglotLimits$EngineLimits.validate(PolyglotLimits.java:319)
	at com.oracle.truffle.polyglot.PolyglotEngineImpl.addContext(PolyglotEngineImpl.java:773)
	at com.oracle.truffle.polyglot.PolyglotEngineImpl.createContext(PolyglotEngineImpl.java:1444)
	at org.graalvm.polyglot.Context$Builder.build(Context.java:1598)
```
</details>

**Comment by oubidar-Abderrahim** [comment]
Could you please provide the steps to reproduce this issue?

**Comment by jdimona** [comment]
Hi, Abderrahim. Thanks for looking into this.

```
// Create an engine.
Engine engine = Engine.newBuilder().build();

// Create a context from the engine with a statement limit.
ResourceLimits limits = ResourceLimits.newBuilder()
        .statementLimit(1000, s -> !s.isInternal())
        .build();
Context context = Context.newBuilder("js")
        .engine(engine)
        .resourceLimits(limits)
        .build();

// Create another context from the engine with no statement limit. At this point the exception is thrown.
Context anotherContext = Context.newBuilder("js")
        .engine(engine)
        .build();
```

**Comment by oubidar-Abderrahim** [comment]
I meant what was the command you used to run the example? did you provide any extra flags to the JVM?

**Comment by chumer** [comment]
Thanks for reporting this. This is indeed a bug, we will look into fixing this.
This should work as a workaround:
```
        Engine engine = Engine.newBuilder().build();
        Predicate<Source> notInternal = (s) -> !s.isInternal();
        ResourceLimits limits = ResourceLimits.newBuilder().statementLimit(1000, notInternal).build();
        Context context = Context.newBuilder("js").engine(engine).resourceLimits(limits).build();
        context.close(true);
        limits = ResourceLimits.newBuilder().statementLimit(Long.MAX_VALUE, notInternal).build();
        Context newContext = Context.newBuilder("js").engine(engine).resourceLimits(limits).build();
```
It is equivalent what would be done under the hood if not specifying the limit would work.

**Comment by jdimona** [comment]
Excellent, thank you.

**Comment by oubidar-Abderrahim** [comment]
Should be solved by commit 1e6b914
